pyart/retrieve/kdp_proc.py

```diff
@@ -21,11 +33,994 @@
+        
+    kdp = np.zeros(psidp_o.shape)*np.nan
+    kdp = np.ma.masked_array(kdp, fill_value = fill_value)
+    
+    kdp_stdev = np.zeros(psidp_o.shape)*np.nan
+    kdp_stdev = np.ma.masked_array(kdp_stdev, fill_value = fill_value)
+    
+    phidp_rec = np.zeros(psidp_o.shape)*np.nan
+    phidp_rec = np.ma.masked_array(phidp_rec, fill_value = fill_value)
+
+    for i,l in enumerate(list_est):
+        kdp[i,0:len(l[0])] = l[0]
+        kdp_stdev[i,0:len(l[1])] = l[1]
+        phidp_rec[i,0:len(l[2])] = l[2]
+    
+    # Mask the estimated Kdp and reconstructed Phidp with the mask of original
+    # psidp
+    if isinstance(psidp_o, np.ma.masked_array):
+        masked = psidp_o.mask
+        kdp = np.ma.array(kdp, mask = masked, fill_value = fill_value)
+        kdp_stdev =  np.ma.array(kdp_stdev, mask = masked, fill_value = fill_value)
+        phidp_rec =  np.ma.array(phidp_rec, mask = masked, fill_value = fill_value)
+
+    # create specific differential phase field dictionary and store data
+    kdp_dict = get_metadata(kdp_field)
+    kdp_dict['data'] = kdp
+    kdp_dict['valid_min'] = -1.0
+    
+    # create reconstructed differential phase field dictionary and store data
+    phidpr_dict = get_metadata(phidp_field)
+    phidpr_dict['data'] = phidp_rec
+    phidpr_dict['valid_min'] = 0.0    
+    
+     # create specific phase stdev field dictionary and store data
+    kdp_stdev_dict = {}
+    kdp_stdev_dict['units'] = 'degrees/km'
+    kdp_stdev_dict['standard_name'] = 'estimated KDP stdev'    
+    kdp_stdev_dict['long_name'] = 'Estimated stdev of spec. diff. phase (KDP)'
+    kdp_stdev_dict['coordinates'] = 'elevation azimuth range'
+    kdp_stdev_dict['data'] = kdp_stdev
+    kdp_stdev_dict['valid_min'] = 0.0    
+    
+    if parallel:
+        pool.close()
+    
+    return kdp_dict, kdp_stdev_dict, phidpr_dict
+
+
+
+def _kdp_estimation_backward_fixed(psidp_in, rcov, pcov_scale, f, f_transposed, h_plus, c1, c2, b1, b2, kdp_th, mpsidp):
+    """
+    Processing one profile of Psidp and estimating Kdp and Phidp
+    with the KFE algorithm described in Schneebeli et al, 2014
+    IEEE_TGRS. This routine estimates Kdp in the backward
+    direction given a set of matrices that define the Kalman
+    filter.
+    
+    Parameters
+    ----------
+    psidp_in : ndarray
+        one-dimensional vector of length -nrg- containining the input psidp
+        [degrees]
+    rcov : 3x3 float array##
+        Measurement error covariance matrix
+    pcov  : 4x4 float array
+        Scaled state transition error covariance matrix
+    F : 4x4 float array 
+        Forward state prediction matrix [4x4]
+    F_transposed: 4x4 float array 
+        Transpose of F
+    H_plus : 4x3 float array
+        Measurement prediction matrix [4x3]
+    c1, c2,b1,b2: floats
+        the values of the intercept of the relation c  = b*Kdp - delta.
+        This relation uses b1, c1 IF kdp is lower than a kdp_th and b2, c2 
+        otherwise kdp_th
```

Comment:
I added the following doc:

    kdp_th: float
        the kdp threshold which separates the two Kdp - delta regime
        i.e. the power law relating delta to Kdp will be different if Kdp is
        larger or smaller than kdp_th
    mpsidp: float
        final observed value of psidp along the radial (usually also the max value),
        needed for inverting the psidp vector